sunpy/net/vso/vso.py

```diff
@@ -909,3 +911,43 @@ def _can_handle_query(cls, *query):
     @classmethod
     def _attrs_module(cls):
         return 'vso', 'sunpy.net.vso.attrs'
+
+    @classmethod
+    def register_values(cls):
+        try:
+            return {cls: cls.get_vso_values()}
+        except Exception as e:
+            warnings.warn(f'Fetching VSO attrs failed with {e}')
+            return {cls: {}}
+
+    def get_vso_values():
+        """
+        Reads the VSO keywords and returns a dict with them for the attrs we support.
+        """
+        from sunpy.net import attrs as a
+
+        # Keywords we are after
+        keywords = ["+detector", "+instrument", "+source", "+provider", "+physobs"]
+
+        # Construct and format the request
+        keyword_info = {}
+        url = "https://vso1.nascom.nasa.gov/cgi-bin/registry_json.cgi"
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        for keyword in keywords:
+            data = urlencode({'fields': f"['{keyword}']".replace("'", '"')}).encode('ascii')
+            req = Request(url=url, data=data, headers=headers)
+            response = urlopen(req)
+            keyword_info[keyword.replace("+", "")] = json.loads(response.read())
+
+        # Now to traverse the return and create attrs out of them.
+        attrs = {}
+        for key, value in keyword_info.items():
+            try:
+                attr = getattr(a, key.capitalize())
+            except AttributeError:
+                attr = getattr(a.vso, key.capitalize())
```

Comment:
You could avoid the try like this?
```suggestion
            attr = getattr(a, key.capitalize(), None)
            if attr is None:
                attr = getattr(a.vso, key.capitalize())     
```

Reply:
Ah I didn't think to check if getattr has a default if it doesn't exist.